packages/headless/src/controllers/case-input/headless-case-input.ts

```diff
@@ -0,0 +1,152 @@
+import {CaseAssistEngine} from '../../app/case-assist-engine/case-assist-engine';
+import {
+  CaseAssistConfigurationSection,
+  CaseFieldSection,
+  CaseInputSection,
+  ConfigurationSection,
+  DocumentSuggestionSection,
+} from '../../state/state-sections';
+import {buildController, Controller} from '../controller/headless-controller';
+import {
+  caseAssistConfiguration,
+  caseField,
+  caseInput,
+  configuration,
+  documentSuggestion,
+} from '../../app/reducers';
+import {Schema, StringValue} from '@coveo/bueno';
+import {validateOptions} from '../../utils/validate-payload';
+import {loadReducerError} from '../../utils/errors';
+import {fetchDocumentSuggestions} from '../../features/document-suggestion/document-suggestion-actions';
+
+export interface CaseInputOptions {
+  field: string;
+}
+
+export interface CaseInputProps {
+  options: CaseInputOptions;
+}
+
+function validateCaseInputOptions(
+  engine: CaseAssistEngine,
+  options: Partial<CaseInputOptions> | undefined
+) {
+  const schema = new Schema<CaseInputOptions>({
+    field: new StringValue({
+      required: true,
+      emptyAllowed: false,
+    }),
+  });
+  validateOptions(engine, schema, options, 'buildCaseInput');
+}
+
+export interface UpdateFetchOptions {
+  caseClassifications?: boolean;
+  documentSuggestions?: boolean;
+}
+
+/**
+ * The `CaseInput` controller is responsible for setting and retrieving the value of a single field from the case creation form and optionally trigger Case Assist API requests.
+ */
+export interface CaseInput extends Controller {
+  /**
+   * Sets the value of the specified field.
+   *
+   * @param value - The case input value to set.
+   * @param updatesToFetch - A set of flags dictating whether to fetch case assist data after updating the input value.
+   */
+  update(value: string, updatesToFetch?: UpdateFetchOptions): void;
+  /**
+   * A scoped and simplified part of the headless state that is relevant to the `CaseInput` controller.
+   */
+  state: CaseInputState;
+}
+
+export interface CaseInputState {
+  /**
+   * The value of the case input.
+   */
+  value: string;
+}
+
+/**
+ * Creates a `Case Input` controller instance.
+ *
+ * @param engine - The headless engine.
+ * @param props - The configurable `CaseInput` controller properties.
+ * @returns A `CaseInput` controller instance.
+ */
+export function buildCaseInput(
+  engine: CaseAssistEngine,
+  props: CaseInputProps
+): CaseInput {
+  if (!loadCaseInputReducers(engine)) {
+    throw loadReducerError;
+  }
+
+  const controller = buildController(engine);
+  const {dispatch} = engine;
+  const getState = () => engine.state;
+
+  validateCaseInputOptions(engine, props.options);
+
+  const fieldName = props.options.field;
+
+  dispatch(
+    updateCaseInput({
+      fieldName: fieldName,
+      fieldValue: '',
+    })
+  );
+
+  return {
+    ...controller,
+
+    update(
+      value: string,
+      updatesToFetch?: {
```

Comment:
What is the scenario where this `updatesToFetch` should be undefined ? Or that you would not want to get case classification or new document suggestions ?

Said differently, what should be the default ? 
Because that's pretty important for ease of use and discoverability of the lib.

Reply:
You could simply want to update the case input value in the state without triggering the fetch. For example if your case information form is on one step and the document suggestion are on the next. you might just trigger the fetch later so you would call `update(value)`.
But if you have a both visible at the same time and you want the suggestions to update in response to changes, set an updatesToFetch flag with `update(value, {documentSuggestions: true})`

Reply:
Could another approach be to provide `updateCaseClassifications` and `updateDocumentSuggestions` methods on the controller?

Reply:
The Case Field controller can also dispatch the `fetchCaseClassifications` and `fetchDocumentSuggestions` actions. Adding `updateCaseClassifications` and `updateDocumentSuggestions` on both Case Input and Case Field controllers will be confusing to developers in my opinion. A developer might wonder if `CaseInput.updateCaseClassifications` and `CaseField.updateCaseClassifications` are doing the same thing? When should I use one over the other?

I think passing the flags to trigger the right calls is a good balance between ease of use and flexibility.

Reply:
I agree and this was my reasoning as well. The fetch actions triggered by the case-input controller should be linked to the main functions of a case-input, namely, storing and updating a value.